server/controllers/user.controller.js

```diff
@@ -44,37 +31,33 @@ function create(req, res, next) {
  * @returns {User}
  */
 function update(req, res, next) {
-  const user = req.user;
-  user.username = req.body.username;
-  user.mobileNumber = req.body.mobileNumber;
-
-  user.save()
-    .then(savedUser => res.json(savedUser))
-    .catch(e => next(e));
-}
-
-/**
- * Get user list.
- * @property {number} req.query.skip - Number of users to be skipped.
- * @property {number} req.query.limit - Limit number of users to be returned.
- * @returns {User[]}
- */
-function list(req, res, next) {
-  const { limit = 50, skip = 0 } = req.query;
-  User.list({ limit, skip })
-    .then(users => res.json(users))
-    .catch(e => next(e));
-}
-
-/**
- * Delete user.
- * @returns {User}
- */
-function remove(req, res, next) {
-  const user = req.user;
-  user.remove()
-    .then(deletedUser => res.json(deletedUser))
-    .catch(e => next(e));
-}
-
-export default { load, get, create, update, list, remove };
+  const user = req.userLoaded;
+  const username = req.body.username;
+  // We gotta check a few things:
+  // First we make sure we are the actual user we are modifying.
+  if(!req.user || user._id != req.user._id) {
```

Comment:
This entire update method feels a bit fragile. Should make a few tasks to expand the tests around it.